File: 03_Spring/demo/jv_spring/src/main/java/com/springbook/biz/polymorphism/speaker/AppleSpeaker.java
package com.springbook.biz.polymorphism.speaker;

public class AppleSpeaker implements Speaker {
    public AppleSpeaker() {
        System.out.println("===> AppleSpeaker 객체 생성");
    }

    @Override
    public void volumeUp() {
        System.out.println("AppleSpeaker --- Volume Up");
    }

    @Override
    public void volumeDown() {
        System.out.println("AppleSpeaker --- Volume Down");
    }
}

File: 03_Spring/kotlin_spring/src/main/kotlin/com/example/kotlin_spring/polymorphism/speaker/AppleSpeaker.kt
package com.example.kotlin_spring.polymorphism.speaker

class AppleSpeaker: Speaker {

    init {
        println("AppleSpeaker 객체 생성..")
    }

    override fun volumeUp() {
        println("AppleSpeaker Volume Up")
    }

    override fun volumeDown() {
        println("AppleSpeaker Volume Down")
    }
}